### PR TITLE
Migrate to Ratatui

### PR DESCRIPTION
**Pull request by joshka**
Ratatui is a replacement for tui-rs that is actively maintained.
See https://ratatui.rs/ for more info.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -15,16 +15,16 @@ edition = "2021"
...
+crossterm = { version = "0.27.0", features = ["use-dev-tty"] }
 anyhow = "1.0"
 clap = { version = "4.2", features = ["derive"] }
 tempfile = "3.5"
 regex = "1.8"
 csv-sniffer = "0.3.1"
 
 [target.'cfg(windows)'.dependencies]
-crossterm = "0.25"
+crossterm = "0.27.0"
```
Comment (YS-L) [inline Cargo.toml:27]: Is it possible keep this pinned at 0.25 for windows? This is due to an on-going issue in crossterm. More info: https://github.com/YS-L/csvlens/issues/30

**Comment by YS-L** [comment]
Thank you @joshka!

File: Cargo.toml (hunk 1 above)
Comment (joshka) [reply Cargo.toml:27]: Fixed in https://github.com/YS-L/csvlens/pull/42/commits/3ce170a62be29f47121a85496bff5a9f6e543112

Usually apps can filter to only include `key_event.kind == Press` to workaround this.
Is there a need to handle release events in this app? I didn't see anything obvious.

See https://github.com/ratatui-org/ratatui/issues/347 for more info.

Comment (YS-L) [reply Cargo.toml:27]: Nice! I think this should work. Thanks for the fix.

**Comment by joshka** [comment]
Not a problem. I've added this to the app showcase at https://ratatui.rs/showcase/apps/#csvlens
I redid your example demo using VHS to be a bit smaller and use a larger font size. See https://github.com/ratatui-org/website/pull/329 for details.

**Comment by YS-L** [comment]
@joshka That's awesome! Thanks for adding csvlens to the page.

**Pull request merged**